src/mbgl/tile/vector_tile.cpp

```diff
@@ -15,9 +15,17 @@ class VectorTileLayer;
 
 using packed_iter_type = protozero::iterator_range<protozero::pbf_reader::const_uint32_iterator>;
 
+struct VectorTileLayerData {
```

Comment:
This should also have a `shared_ptr<const std::string> data` member, to ensure that the data backing the `tags_iter` and `geometry_iter` is retained.